feature/interface/aggregate/otg_tests/aggregate_forwarding_viable_test/aggregate_forwarding_viable_test.go

```diff
@@ -0,0 +1,596 @@
+	dstEth.Ipv4Addresses().Add().SetName(ateDst.Name + ".IPv4").SetAddress(ateDst.IPv4).SetGateway(dutDst.IPv4).SetPrefix(int32(ateDst.IPv4Len))
+	dstEth.Ipv6Addresses().Add().SetName(ateDst.Name + ".IPv6").SetAddress(ateDst.IPv6).SetGateway(dutDst.IPv6).SetPrefix(int32(ateDst.IPv6Len))
+
+	tc.ate.OTG().PushConfig(t, tc.top)
+	tc.ate.OTG().StartProtocols(t)
+}
+
+// incrementMAC uses a mac string and increments it by the given i
+func incrementMAC(mac string, i int) (string, error) {
+	macAddr, err := net.ParseMAC(mac)
+	if err != nil {
+		return "", err
+	}
+	convMac := binary.BigEndian.Uint64(append([]byte{0, 0}, macAddr...))
+	convMac = convMac + uint64(i)
+	buf := new(bytes.Buffer)
+	err = binary.Write(buf, binary.BigEndian, convMac)
+	if err != nil {
+		return "", err
+	}
+	newMac := net.HardwareAddr(buf.Bytes()[2:8])
+	return newMac.String(), nil
+}
+
+// generates a list of random tcp ports values
+func generateRandomPortList(count int) []int32 {
+	a := make([]int32, count)
+	for index := range a {
+		a[index] = int32(rand.Intn(65536-1) + 1)
+	}
+	return a
+}
+
+// normalize normalizes the input values so that the output values sum
+// to 1.0 but reflect the proportions of the input.  For example,
+// input [1, 2, 3, 4] is normalized to [0.1, 0.2, 0.3, 0.4].
+func normalize(xs []uint64) (ys []float64, sum uint64) {
+	for _, x := range xs {
+		sum += x
+	}
+	ys = make([]float64, len(xs))
+	for i, x := range xs {
+		ys[i] = float64(x) / float64(sum)
+	}
+	return sort.Float64Slice(ys), sum
+}
+
+// portWants converts the nextHop wanted weights to per-port wanted
+// weights listed in the same order as atePorts.
+// portWantsViable -> [1/n, 1/n, ...]
+func (tc *testArgs) portWantsViable(t *testing.T) []float64 {
+	weights := []float64{}
+	numPorts := len(tc.dutPorts[1:])
+	for range tc.dutPorts[1:] {
+		weights = append(weights, 1/float64(numPorts))
+	}
+	return sort.Float64Slice(weights)
+}
+
+// portWants converts the nextHop wanted weights to per-port wanted
+// weights listed in the same order as atePorts.
+// portWantsNotViable -> [0, 1/(n-1), 1/(n-1), ...]
+func (tc *testArgs) portWantsNotViable(t *testing.T) []float64 {
+	weights := []float64{}
+	// Forwarding viable flag is set as false for one of the port in the port channel
+	numPorts := len(tc.dutPorts[2:])
+	weights = append(weights, 0)
+	for range tc.dutPorts[2:] {
+		weights = append(weights, 1/float64(numPorts))
+	}
+	return sort.Float64Slice(weights)
+}
+
+// debugATEFlows logs detailed tracking information on traffic flows and find if there is any loss pct in the flow.
+func debugATEFlows(t *testing.T, ate *ondatra.ATEDevice, flow gosnappi.Flow, lp linkPairs) {
+
+	recvMetric := gnmi.Get(t, ate.OTG(), gnmi.OTG().Flow(flow.Name()).State())
+	txPackets := float32(recvMetric.GetCounters().GetOutPkts())
+	rxPackets := float32(recvMetric.GetCounters().GetInPkts())
+	lostPackets := txPackets - rxPackets
+	lossPct := lostPackets * 100 / txPackets
```

Comment:
Use floats.

Reply:
txPackets and rxPackets are floats and corrected the Fatalf description below to %f